python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -5,209 +5,313 @@
 
-    def set(self, name: str, value: Any) -> Self:
-        """
-        Set a user config option.
+class Scheduler(str, enum.Enum):
+    """
+    The scheduler to use for the streaming executor.
 
-        Nested dictionary keys should be separated by periods.
-        For example::
+    * ``Scheduler.SYNCHRONOUS`` : Use the synchronous scheduler.
+    * ``Scheduler.DISTRIBUTED`` : Use the distributed scheduler.
+    """
 
-            >>> options = options.set("parquet_options.chunked", False)
+    SYNCHRONOUS = "synchronous"
+    DISTRIBUTED = "distributed"
 
-        Parameters
-        ----------
-        name
-            Period-separated config name.
-        value
-            New config value.
-        """
-        options = config_options = copy.deepcopy(self.config_options)
-        keys = name.split(".")
-        for k in keys[:-1]:
-            assert isinstance(options, dict)
-            if k not in options:
-                options[k] = {}
-            options = options[k]
-        options[keys[-1]] = value
-        return type(self)(config_options)
-
-    def get(self, name: str, *, default: Any = None) -> Any:
-        """
-        Get a user config option.
 
-        Nested dictionary keys should be separated by periods.
-        For example::
+class ShuffleMethod(str, enum.Enum):
+    """
+    The method to use for shuffling data between workers with the streaming executor.
 
-            >>> chunked = config_options.get("parquet_options.chunked")
+    * ``ShuffleMethod.TASKS`` : Use the task-based shuffler.
+    * ``ShuffleMethod.RAPIDSMPF`` : Use the rapidsmpf scheduler.
 
-        Parameters
-        ----------
-        name
-            Period-separated config name.
-        default
-            Default return value.
+    In :class:`StreamingExecutor`, the default of ``None`` will attempt to use
+    ``ShuffleMethod.RAPIDSMPF``, but will fall back to ``ShuffleMethod.TASKS``
+    if rapidsmpf is not installed.
+    """
 
-        Returns
-        -------
-        The user-specified config value, or `default`
-        if the config is not found.
-        """
-        options = self.config_options
-        keys = name.split(".")
-        for k in keys[:-1]:
-            assert isinstance(options, dict)
-            options = options.get(k, {})
-        return options.get(keys[-1], default)
+    TASKS = "tasks"
+    RAPIDSMPF = "rapidsmpf"
 
-    def __hash__(self) -> int:
-        """Hash a ConfigOptions object."""
-        try:
-            return self._hash_value
-        except AttributeError:
-            self._hash_value = hash(json.dumps(self.config_options))
-            return self._hash_value
-
-    @staticmethod
-    def validate(config: dict) -> dict:
-        """
-        Validate a configuration-option dictionary.
 
-        Parameters
-        ----------
-        config
-            GPUEngine configuration options to validate.
+STREAMING_EXECUTOR_PARQUET_DEFAULTS = {
+    "chunked": False,
+}
```

Comment:
@rjzamora strange question: but do you remember why I added this?

This is what's causing us to miss covering [these lines](https://github.com/rapidsai/cudf/blob/79b23be786420824ea6dcaefe3b8956ca3fc33f9/python/cudf_polars/cudf_polars/dsl/ir.py#L632-L634) in `ir.py`. On `branch-25.06`, `config_options.get("parquet_options.chunked", True)` resolves to `True` because `parquet_options.chunked` is None. That seems to be the only test covering this, and so we fail.

This block defines the per-executor defaults for `ParquetOptions`. I'm probably misremembering, but I thought you mentioned `chunked=False` being desirable for the streaming executor.

Reply:
On branch-25.06 we change the `"parquet_options.chunked"` config to `False` when we construct a `SplitScan` node (i.e. when we decide to create new IR nodes to read  _partial_ files). When we don't run into large files, the streaming executor doesn't bother changing any defaults, so we still use/cover this chunked-reader code path.

Reply:
I think it makes sense to remove this default for the streaming executor. I'm pretty sure you are using `replace` to disable the chunked reader in the split-file case.

Reply:
Thanks for clearing that up. Sounds like I mistook the specific guidance to use `chunked=False` in that specific case to mean use `chunked=False` always with `executor="streaming"`.

Reply:
>Sounds like I mistook the specific guidance to use chunked=False in that specific case to mean use chunked=False always with executor="streaming".

I probably did say that we could disallow chunked=True with the streaming executor if it makes things "easier." In this case, it looks like it just complicates things a bit.